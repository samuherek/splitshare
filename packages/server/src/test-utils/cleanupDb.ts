import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
