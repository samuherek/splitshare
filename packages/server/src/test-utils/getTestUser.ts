import { Connection } from 'typeorm';
import { User } from '../features/user/entity';

export const testUser = {
  email: 'user@email.com',
};

export async function getTestUser(conn: Connection) {
  const user = await conn.getRepository(User).findOne(testUser);

  if (!user) {
    throw new Error('No Test user found');
  }

  return user;
}
