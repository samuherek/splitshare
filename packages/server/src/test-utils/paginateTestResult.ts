import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
