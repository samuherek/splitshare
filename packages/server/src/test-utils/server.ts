import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { getSchema } from '../schema';

export default async function createTestServer(ctx: any) {
  const schema = await getSchema();

  const server = new ApolloServer({
    schema,
    mockEntireSchema: false,
    mocks: true,
    context: () => ctx,
  });

  return createTestClient(server);
}
