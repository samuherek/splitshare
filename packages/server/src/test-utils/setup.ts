// @ts-ignore
import * as tryToCatch from 'try-to-catch';
import { Connection } from 'typeorm';
import { connectTestDatabase } from './testConnection';

(async () => {
  const [error, conn]: [Error, Connection] = await tryToCatch(
    connectTestDatabase,
    { drop: true }
  );

  if (error) {
    throw new Error(error.message);
  }
})();
