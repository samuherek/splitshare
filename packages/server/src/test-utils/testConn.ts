import { createConnection } from 'typeorm';

export const testConn = (drop: boolean = false) => {
  return createConnection({
    database: 'splitshare-test',
    dropSchema: drop,
    host: 'localhost',
    name: 'default',
    password: '',
    port: 5432,
    entities: [__dirname + '/../entity/**/*.*'],
    migrations: [__dirname + '/../migration/**/*.*'],
    subscribers: [__dirname + '/../subscriber/**/*.*'],
    synchronize: drop,
    type: 'postgres',
    username: 'samuherek',
  });
};
