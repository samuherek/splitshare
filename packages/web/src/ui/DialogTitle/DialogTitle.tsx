// @flow
import clsx from 'clsx';
import React from 'react';
import styled from 'styled-components';

type Props = {
  className?: string;
  children?: any;
};

export const classes = {
  root: 'DialogTitle',
};

const WrapStyled = styled.div`
  flex: 0 0 auto;
  margin: 0;
  padding: 16px 24px 8px;

  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  span {
    line-height: 1.2;
  }
`;

function DialogTitle({ className, ...props }: Props) {
  return <WrapStyled className={clsx(className, classes.root)} {...props} />;
}

export default DialogTitle;
