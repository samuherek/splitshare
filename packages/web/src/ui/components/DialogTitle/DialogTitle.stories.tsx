import React from 'react';
import DialogTitle from './DialogTitle';

export default {
  title: 'Dialog/Title',
  component: DialogTitle,
};

export const Base = () => {
  return (
    <DialogTitle>
      <div>Dialog Title</div>
    </DialogTitle>
  );
};
