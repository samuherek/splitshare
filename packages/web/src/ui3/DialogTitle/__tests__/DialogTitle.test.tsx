import React from 'react';
import { render } from '../../../test-utils/test-utils';
import DialogTitle from '../DialogTitle';

test('<DialogTitle /> should match snpashot', () => {
  const { container } = render(<DialogTitle>Content</DialogTitle>);
  const title = container.firstChild;
  expect(title).toMatchSnapshot();
});
