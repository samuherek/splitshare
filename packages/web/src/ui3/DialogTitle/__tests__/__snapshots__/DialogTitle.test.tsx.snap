// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogTitle /> should match snpashot 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin: 0;
  padding: 16px 24px 8px;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6,
.c0 p,
.c0 span {
  line-height: 1.2;
}

<div
  class="c0 DialogTitle"
>
  Content
</div>
`;
